src/vector/backends/object.py

```diff
@@ -742,31 +734,11 @@ def _wrap_result(
 
         elif (
             len(returns) == 2
-            and isinstance(returns[0], type)
-            and issubclass(returns[0], Azimuthal)
-            and returns[1] is None
-        ):
-            azcoords = _coord_object_type[returns[0]](result[0], result[1])
-            return cls.ProjectionClass2D(azimuthal=azcoords)
-
-        elif (
-            len(returns) == 2
-            and isinstance(returns[0], type)
-            and issubclass(returns[0], Azimuthal)
-            and isinstance(returns[1], type)
-            and issubclass(returns[1], Longitudinal)
-        ):
-            azcoords = _coord_object_type[returns[0]](result[0], result[1])
-            lcoords = _coord_object_type[returns[1]](result[2])
-            return cls.ProjectionClass3D(azimuthal=azcoords, longitudinal=lcoords)
-
-        elif (
-            len(returns) == 3
+            or (len(returns) == 3 and returns[2] is None)
             and isinstance(returns[0], type)
             and issubclass(returns[0], Azimuthal)
             and isinstance(returns[1], type)
             and issubclass(returns[1], Longitudinal)
-            and returns[2] is None
         ):
```

Comment:
I had to manually make these changes, but pretty impressed Ruff noticed it could be made. :)